packages/commonwealth/server/util/cosmosCache.ts

```diff
@@ -27,19 +27,18 @@ export function calcCosmosLCDCacheKeyDuration(req, res, next) {
 
   if (proposalStatus) {
     if (activeProposalCodes.some((c) => c === +proposalStatus)) {
-      // ACTIVE PROPOSALS: short cache
-      duration = oneBlock;
+      // ACTIVE PROPOSALS: 5 minute cache
+      duration = 60 * 5;
     } else if (completedProposalCodes.some((c) => c === +proposalStatus)) {
-      // COMPLETED PROPOSALS: cache 15 minutes
-      duration = 60 * 15;
+      // COMPLETED PROPOSALS: cache 30 minutes
+      duration = 60 * 30;
     }
-    // specific proposal request:
   } else if (url.includes('/proposals/')) {
-    // LIVE VOTES/TALLY/DEPOSITS: 5 minutes
-    duration = 60 * 5;
+    // specific proposal request: long-term cache (1 year)
+    duration = 60 * 60 * 24 * 365;
```

Comment:
Is it safe to cache votes specifically for such a long time, as votes get actively changed?

Reply:
Added a more targeted case for active proposal votes, tally, and deposits. Cache set at 10 minutes.

Also, I noticed Proposal base data includes a `final_tally`. So I set cache for an individual proposal at 7 days (instead of 1 year). This will ensure the active proposal's `final_tally` gets updated after it becomes a completed proposal. It would be nice to have a permanent cache once completed, but that is a different caching strategy (we can't tell whether active or completed with the request).

Reply:
(tested live proposals with http://localhost:8080/csdk)

Reply:
@mhagel So -- what's our final list of caching endpoints and TTLs? Would like to confirm each of the settings before we ship.

Reply:
This describes it pretty well.

https://github.com/hicommonwealth/commonwealth/blob/400d8d0b6b03d6e2f5fa42d9d7d76e81eb9cbeea/packages/commonwealth/server/util/cosmosCache.ts#L53-L80


It would be good to put together a list of expected cache keys and their durations. That might take me a minute.

Reply:
```
cacheKey | duration (seconds)

/cosmosAPI/osmosis_{"jsonrpc":"2.0","id":393358497138,"method":"status","params":{}} :  6
/cosmosAPI/osmosis_/cosmos.staking.v1beta1.Query/Pool :  432000
/cosmosAPI/osmosis_/cosmos.staking.v1beta1.Query/Params :  432000
/cosmosAPI/osmosis_{"height":"10520482"} :  6
/cosmosAPI/osmosis_{"path":"/cosmos.gov.v1beta1.Query/Proposals","data":"0803","prove":false} :  1800
/cosmosAPI/osmosis_{"path":"/cosmos.gov.v1beta1.Query/Proposals","data":"0803220a0a080000000000000087","prove":false} :  1800
/cosmosAPI/osmosis_{"path":"/cosmos.gov.v1beta1.Query/Proposals","data":"0805","prove":false} :  1800
/cosmosAPI/osmosis_{"path":"/cosmos.gov.v1beta1.Query/Proposals","data":"0804","prove":false} :  1800
/cosmosAPI/osmosis_{"path":"/cosmos.gov.v1beta1.Query/Proposals","data":"0802","prove":false} :  300
/cosmosAPI/osmosis_{"path":"/cosmos.gov.v1beta1.Query/Proposals","data":"0801","prove":false} :  300
/cosmosAPI/osmosis_{"path":"/cosmos.gov.v1beta1.Query/Votes","data":"08b004","prove":false} :  600
/cosmosAPI/osmosis_{"path":"/cosmos.gov.v1beta1.Query/TallyResult","data":"08b004","prove":false} :  600
/cosmosAPI/osmosis_{"path":"/cosmos.gov.v1beta1.Query/Deposits","data":"08b004","prove":false} :  600

```

Reply:
@jnaviask See above. Hopefully these are clear enough.

These are the equivalent of /deposit, /tallying, /voting params requests. I'm just not sure which is which.
```
/cosmosAPI/osmosis_{"path":"/cosmos.gov.v1beta1.Query/Params","data":"0a076465706f736974","prove":false} |  432000
/cosmosAPI/osmosis_{"path":"/cosmos.gov.v1beta1.Query/Params","data":"0a0874616c6c79696e67","prove":false}  | 432000
/cosmosAPI/osmosis_{"path":"/cosmos.gov.v1beta1.Query/Params","data":"0a06766f74696e67","prove":false} | 432000
```

I confirmed that the "0a076465706f736974" numbers are consistent across chains, so these are "voting" encoded, etc.

Reply:
This looks fine to me, with the only caveat that a 10-minute lag between a user voting + the vote appearing in the data might be confusing -- the user might worry their vote didn't go through.

Reply:
Good point. The good news is that In the normal user flow of voting, their vote appears immediately after success:

<img width="723" alt="Screenshot 2023-07-17 at 3 08 20 PM" src="https://github.com/hicommonwealth/commonwealth/assets/9438198/2ce846a3-702f-411e-b7fc-fbc3316a1b9c">

This particular UI update is separate from the cached /votes requests.

But your concern is correct that it won't be there if they refresh the page or go back to proposals. Their vote won't show up for ten minutes. We currently have this set at 5 minutes, so it is already doing this. However, maybe we should just cache this very short, if at all.